src/defaults.rs

```diff
@@ -0,0 +1,30 @@
+use crate::bindings::*;
```

Comment:
Why would we want to do this here and not in `esp-idf-svc`?

Reply:
You cannot implement `Default` in another crate, so it needs to be defined here.

Reply:
No I mean given that we have [a type-safe wrapper for Wifi already](https://github.com/esp-rs/esp-idf-svc/blob/master/src/wifi.rs#L268), why would you need these defaults here? Or do you plan on implementing another set of type-safe wrappers besides `esp-idf-svc`? I would be supper happy if you could help us improve `esp-idf-svc` instead. :)

One more point: we have discussed with Espressif, that the best possible solution is to have C functions that implement these defaults - in addition to their C macros that we cannot consume. Timelines unclear though.

Reply:
I have implemented a different wrapper in one of my projects which used my own `bindgen ` crate before these crates existed. This would enable me to switch to this crate. Also, this could still be used in `esp-idf-svc` which currently does the same thing as in this `Default` implementation.

> I would be supper happy if you could help us improve `esp-idf-svc` instead. :)

I will try to have a more detailed look at the `esp-idf-svc` crate. I think my approach is a more high level API so it could potentially depend on the `esp-idf-svc` crate.

Reply:
Ok so I'm all for switching to Rust enums if you can make it work in esp idf svc and esp idf hal. As for the default method, in light of the above idea of having it centralized in ESP IDF and consumable via CAPI call, perhaps we should wait with it a bit?

Reply:
By the way, `EspWifi` is pretty high level if you put aside the fact that it is mostly async (for which there are good reasons and you can workaround that). Not sure how much higher level you can get?

Reply:
> perhaps we should wait with it a bit?

I don't think it makes a difference. `default` can easily be changed to use the CAPI call once it is available.

> Not sure how much higher level you can get?

I'm also not sure. As I said, I implemented my wrapper before these crates existed, so I have also not yet reviewed in detail how `EspWifi` is implemented.